components/stateTracker.js

```diff
@@ -90,6 +90,7 @@ StateTracker.prototype.bind = function() {
     var self = this;
     $(window).on('popstate', function() {
         if (!self.disabled) {
+            _.extend(history.state || {}, {_stateFromHistory: true});
```

Comment:
I assume that nomally application should not depend on the type of stateChange. Following this logic this field should starts with underscore.